plate_linker/plate_linker.py

```diff
@@ -0,0 +1,76 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--, The Wetlab Assistant Development Team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+
+import argparse
+
+
+def plate_linker(metadata_f, primer_f, output_f):
+    """Transfer sample IDs to mapping file by well IDs.
+
+    Parameters
+    ----------
+    metadata_f : file object
+        Input metadata file
+    primer_f : file object
+        Input primer file
+    output_f : file object
+        Output mapping file
+
+    Notes
+    -----
+    This function joins a metadata file and a primer file (barcode sequence
+    template file) by well, into a complete mapping file.
+
+    In the metadata file, the 1st column is the sample ID, the 2nd and 3rd
+    columns are primer plate ID and well ID, and the remaining columns are
+    metadata to be appended.
+
+    In the primer file, the 2nd and 3rd columns are barcode and primer, while
+    the 4th and 5th columns are primer plate ID and well ID.
+
+    In the output file, columns are: sample ID, barcode, primer, primer plate
+    ID, well ID, plus variable number of metadata columns.
+    """
+    # merge column headers of primer and metadata files
+    primcols = primer_f.readline().strip('\r\n').split('\t')
+    metacols = metadata_f.readline().strip('\r\n').split('\t')[3:]
```

Comment:
Given that the input files are very specific, do you think it would be a good idea to have some checking, and throw an error if the columns aren't in place (i.e. plate ID is missing).